src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs

```diff
@@ -31,6 +33,13 @@ public async Task<IActionResult> GetPledges()
         [Route("accounts/{accountId}/pledges")]
         public async Task<IActionResult> CreatePledge(long accountId, [FromBody]CreatePledgeRequest createPledgeRequest)
         {
+            var locationInformation = new List<GetLocationInformationResult>(); 
```

Comment:
I think this probably belongs in the (create pledge) handler and not the controller. In the handler you'd be able to inject the location service directly to get the information.